packages/test/src/test-integration-workflows.ts

```diff
@@ -222,3 +222,75 @@ test('Start of workflow with signal is delayed', async (t) => {
   const startDelay = workflowExecutionStartedEvent?.workflowExecutionStartedEventAttributes?.firstWorkflowTaskBackoff;
   t.is(tsToMs(startDelay), 4678000);
 });
+
+export async function executeEagerActivity(): Promise<string> {
+  // If eager activity dispatch is working, then the task will always be dispatched to the workflow
+  // worker. Otherwise, chances are 50%-50% for either workers. We execute the activity 20 times to
+  // make sure that the workflow worker is really getting the task thanks to eager activity
+  // dispatch, and not out of pure luck.
+  for (let i = 0; i < 20; i++) {
+    const res = await workflow
+      .proxyActivities({ scheduleToCloseTimeout: '5s', allowEagerDispatch: true })
+      .testActivity()
+      .catch(() => 'failed');
+    if (res === 'failed') return res;
+  }
+  return 'success';
+}
+
+test('Worker request Eager Activity Dispatch if possible', async (t) => {
+  const { createWorker, startWorkflow } = helpers(t);
+
+  const activityWorker = await createWorker({
+    activities: {
+      testActivity: () => 'failed',
+    },
+    workflowBundle: undefined,
+    maxConcurrentActivityTaskPolls: 100,
+  });
+  const workflowWorker = await createWorker({
+    activities: {
+      testActivity: () => 'success',
+    },
+    maxConcurrentActivityTaskPolls: 2,
+  });
+  const handle = await startWorkflow(executeEagerActivity);
+  const result = await activityWorker.runUntil(workflowWorker.runUntil(handle.result()));
+  const { events } = await handle.fetchHistory();
+
+  t.is(result, 'success');
+  t.false(events?.some?.((ev) => ev.activityTaskTimedOutEventAttributes));
+  const activityTaskStarted = events?.filter?.((ev) => ev.activityTaskStartedEventAttributes);
+  t.is(activityTaskStarted?.length, 20);
+  t.true(activityTaskStarted?.every((ev) => ev.activityTaskStartedEventAttributes?.attempt === 1));
+});
+
+export async function dontExecuteEagerActivity(): Promise<string> {
+  return (await workflow
+    .proxyActivities({ scheduleToCloseTimeout: '5s', allowEagerDispatch: true })
+    .testActivity()
+    .catch(() => 'failed')) as string;
+}
+
+test('Worker dont request Eager Activity Dispatch if no activity registered', async (t) => {
```

Comment:
Please assert that the activity doesn't have the requestEagerExecution flag set.

Reply:
> Please assert that the activity doesn't have the requestEagerExecution flag set.

Can't do that directly. `requestEagerExecution` only exists on the `ScheduleActivityTaskCommand` message that Core sends back to the server. The flag isn't present on anything after that point, either in WF History or on the Activity Task. Therefore we can only test that it got eagerly dispatched by the side effects of it.

Reply:
In this case, if the activity was eagerly dispatched to the Workflow worker even though it is a Workflow-only worker, then the activity execution will timeout (because tasks are not being polled) or otherwise fail (because no activity is registered under that name). Therefore, if the history shows only one attempt for that activity and no timeout, that can only mean that the activity was not eagerly dispatched; at that point, we can't confirm exactly why it wasn't (e.g. could be that the feature is disabled on the server), which is why I also added the other test, that confirms that Eager Activity Execution works properly in other cases.

Reply:
Yeah, this is fine, I missed this in the first pass.